File: LeetCode June Challenge/BinaryTreeRightSideView.java
import java.util.*;
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}
class Solution {
    List<Integer> res = new ArrayList<>();
    private void helper(TreeNode root, int level){
        if(root == null) return;
        if(res.size() < level) res.add(root.val);
        helper(root.right, level + 1);
        helper(root.left, level + 1);
    }
    public List<Integer> rightSideView(TreeNode root) {
        helper(root, 1);
        return res;
    }
}

public class BinaryTreeRightSideView {
    public static void main(String[] args) {
        
    }
}

File: Year-2022/LeetCode-July-Challenge/BinaryTreeRightSideView.js
function TreeNode(val, left, right) {
    this.val = (val===undefined ? 0 : val)
    this.left = (left===undefined ? null : left)
    this.right = (right===undefined ? null : right)
}
var rightSideView = function(root) {
    if(!root) return [];
    const q = [root];
    const result = [];
    while(q.length){
        const size = q.length;
        result.push(q[size-1].val);
        for(let i=0; i<size; i++){
            const node = q.shift();
            node.left && q.push(node.left);
            node.right && q.push(node.right);
        }
    }
    return result;
};